--- origsrc/cffi-0.9.2/src/cffi/verifier.py	2015-03-13 04:02:09.000000000 -0500
+++ src/cffi-0.9.2/src/cffi/verifier.py	2015-05-05 22:58:45.571477600 -0500
@@ -286,6 +286,8 @@ def _get_so_suffixes():
         # bah, no C_EXTENSION available.  Occurs on pypy without cpyext
         if sys.platform == 'win32':
             suffixes = [".pyd"]
+        elif sys.platform == 'cygwin':
+            suffixes = [".dll"]
         else:
             suffixes = [".so"]
 
--- origsrc/cffi-1.16.0/src/c/_cffi_backend.c	2023-09-28 19:43:07.000000000 +0200
+++ src/cffi-1.16.0/src/c/_cffi_backend.c	2024-05-14 19:25:37.413150200 +0200
@@ -6118,7 +6118,11 @@ static void _my_PyErr_WriteUnraisable(Py
 
     PyErr_Restore(t, v, tb);
     if (s != NULL) {
+#if PY_MINOR_VERSION >= 13
+        PyErr_FormatUnraisable(PyText_AS_UTF8(s));
+#else
         _PyErr_WriteUnraisableMsg(PyText_AS_UTF8(s), NULL);
+#endif
         Py_DECREF(s);
     }
     else
